apps/blockchain/lib/blockchain/block.ex

```diff
@@ -119,6 +119,36 @@ defmodule Blockchain.Block do
     }
   end
 
+  @spec decode_rlp(binary()) :: {:ok, [ExRLP.t()]} | {:error, any()}
+  def decode_rlp("0x" <> hex_data) do
+    try do
+      hex_binary = Base.decode16!(hex_data, case: :mixed)
```

Comment:
I might not be following all the logic through, but is there a benefit to using `Base.decode16!` along with a `try rescue` instead of `Base.decode16` (without the `!`) and just do a `case` on the return value?

Reply:
@germsvel good point. I will change it in my next PR with blockchain test fixes.